### ISSUE TITLE
[Arch Linux] When opening multiple instances of neovim, and keeping them open presence.nvim shows the first opened instance in Discord

### ISSUE THREAD

**Issue by ModProg** [andweeb/presence.nvim#59]
### Description

presence.nvim shows the first opened (and still open) instance of nvim, not the most recently interacted with

**Solution**
I would like some configurable setting as a threshold to switch to the more recently interacted instance. A sane default would be something like 20-30 seconds, so that opening a config file or something  similar for a few seconds does not trigger switching.

### Neovim version

```markdown
NVIM v0.7.0-dev+923-gecec95712
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### OS information

Arch Linux

### Steps to reproduce

1. Setup nvim.presence with default settings
2. Open nvim with a random file (keep this instance running)
3. Open a second instance with a different file
4. Notice how Discord is still showing the file from 2., no matter how long you use the instance of 3. (it does switch if you close the instance of 2.)

### Logs

As reported in #52 (thanks @Gilgames000!):
> Logs from the first instance I opened:
```
[presence.nvim] Setting up plugin for linux
[presence.nvim] Using runtime path: /run/user/1000
[presence.nvim] Using Discord IPC socket path: /run/user/1000/discord-ipc-0
[presence.nvim] Checking Discord IPC socket at /run/user/1000/discord-ipc-0...
[presence.nvim] Using id 33a37ae3-377b-40fe-8529-6e63b70ffce9
[presence.nvim] Completed plugin setup
[presence.nvim] Getting nvim socket paths...
[presence.nvim] Executing command: `netstat -u|grep --color=never "nvim.*/0"`
[presence.nvim] Checked Discord IPC socket, looks good!
[presence.nvim] Handling BufEnter event...
[presence.nvim] Connecting to Discord...
[presence.nvim] Connected to Discord
[presence.nvim] Authorizing with Discord...
[presence.nvim] Handling FocusGained event...
[presence.nvim] Skipping presence update for FocusGained event triggered by tmux...
[presence.nvim] Wrote message to pipe
[presence.nvim] Got nvim socket paths: {}
[presence.nvim] No other remote nvim instances
[presence.nvim] Authorized with Discord for Gilgames
[presence.nvim] Parsed filename foo with no extension
[presence.nvim] Getting project name for /home/gilgames/presence-test1...
[presence.nvim] Not a git repository, skipping...
[presence.nvim] Setting activity for /home/gilgames/presence-test1/foo...
[presence.nvim] No project detected
[presence.nvim] Sync activity to all peers...
[presence.nvim] Setting Discord activity...
[presence.nvim] Wrote message to pipe
[presence.nvim] Set activity in Discord for foo

```

> Then I switched to another tmux session and opened another neovim instance:
```


[presence.nvim] Setting up plugin for linux
[presence.nvim] Using runtime path: /run/user/1000
[presence.nvim] Using Discord IPC socket path: /run/user/1000/discord-ipc-0
[presence.nvim] Checking Discord IPC socket at /run/user/1000/discord-ipc-0...
[presence.nvim] Using id dd6d178c-01e7-4e81-bb0c-14ec72e08018
[presence.nvim] Completed plugin setup
[presence.nvim] Getting nvim socket paths...
[presence.nvim] Executing command: `netstat -u|grep --color=never "nvim.*/0"`
[presence.nvim] Checked Discord IPC socket, looks good!
[presence.nvim] Handling BufEnter event...
[presence.nvim] Connecting to Discord...
[presence.nvim] Connected to Discord
[presence.nvim] Authorizing with Discord...
[presence.nvim] Handling FocusGained event...
[presence.nvim] Skipping presence update for FocusGained event triggered by tmux...
[presence.nvim] Wrote message to pipe
[presence.nvim] Got nvim socket paths: {}
[presence.nvim] No other remote nvim instances
[presence.nvim] Authorized with Discord for Gilgames
[presence.nvim] Parsed filename bar with no extension
[presence.nvim] Getting project name for /home/gilgames/presence-test2...
[presence.nvim] Not a git repository, skipping...
[presence.nvim] Setting activity for /home/gilgames/presence-test2/bar...
[presence.nvim] No project detected
[presence.nvim] Sync activity to all peers...
[presence.nvim] Setting Discord activity...
[presence.nvim] Wrote message to pipe
[presence.nvim] Set activity in Discord for bar
```
> but discord was still showing that I am editing `foo` on the first instance. So then I went back to the first one and closed it, switched back to the second instance and those lines appeared in the logs:
```
[presence.nvim] Handling FocusGained event...
[presence.nvim] Activity already set for /home/gilgames/presence-test2/bar, skipping...
[presence.nvim] Handling FocusGained event...
[presence.nvim] Activity already set for /home/gilgames/presence-test2/bar, skipping...
[presence.nvim] Handling FocusGained event...
[presence.nvim] Activity already set for /home/gilgames/presence-test2/bar, skipping...
[presence.nvim] Handling FocusGained event...
[presence.nvim] Activity already set for /home/gilgames/presence-test2/bar, skipping...
[presence.nvim] Handling FocusGained event...
[presence.nvim] Activity already set for /home/gilgames/presence-test2/bar, skipping...
```

**Comment by andweeb** [comment]
Thanks for re-opening the issue with the bug template @ModProg! Looks like there's an issue with retrieving peer nvim sockets in Arch Linux via `netstat` (also reported by @Gilgames000 in #52). I've updated your comment with the logs provided in the other issue.

The plugin _should_ immediately and automatically update Discord with the most recently read/edited file. It looks like this issue is OS-specific. The following logs from the second instance highlight the scenario:
```
[presence.nvim] Getting nvim socket paths...
[presence.nvim] Executing command: `netstat -u|grep --color=never "nvim.*/0"`
                ...
[presence.nvim] Got nvim socket paths: {}
[presence.nvim] No other remote nvim instances
```
tldr; Rich Presence doesn't update with the latest interacted instance since instances can't see each other.

@ModProg @Gilgames000 Can one (or both) of you run multiple instances of nvim and run the following commands to see whether they work on your system?
* Using the `netstat` command: `netstat -u | grep --color=never "nvim.*/0"`
* Using the `ss` command: `ss -lx | grep "nvim.*/0"`

**Comment by Gilgames000** [comment]
```
~
❯ netstat -u | grep --color=never "nvim.*/0"

~
❯ ss -lx | grep "nvim.*/0"
u_str LISTEN 0      32
                 /tmp/nvim6eZNAi/0 418741            * 0
u_str LISTEN 0      32
                 /tmp/nvimhS7bS6/0 442921            * 0
```

**Comment by ModProg** [comment]
> Sure @andweeb, `netstat` doesn't return anything whereas `ss` does!

same for me!

**Comment by andweeb** [comment]
Nice, thanks for the quick response folks. Looks like Arch Linux has deprecated [net-tools](https://archlinux.org/packages/?name=net-tools) in favor of [iproute2](https://archlinux.org/packages/?name=iproute2) (as noted in [Network Configuration](https://wiki.archlinux.org/title/Network_configuration#net-tools)). It's interesting `netstat` is still available as an executable despite not working as expected.

Just pushed a fix, Rich Presence should be nice and snappy for you Arch Linux folks now :) Of course please re-open this issue if you've updated your plugin and still observe the same issue!

**Comment by Gilgames000** [comment]
It works like a charm, thanks!
